pkg/download/goget/goget.go

```diff
@@ -22,15 +22,23 @@ import (
 // New returns a download protocol by using
 // go get. You must have a modules supported
 // go binary for this to work.
-func New() download.Protocol {
-	return &goget{
-		goBinPath: env.GoBinPath(),
-		fs:        afero.NewOsFs(),
+func New() (download.Protocol, error) {
+	goBin := env.GoBinPath()
+	fs := afero.NewOsFs()
+	mf, err := module.NewGoGetFetcher(goBin, fs)
+	if err != nil {
+		return nil, fmt.Errorf("go get fetcher can't be initialised with gobinpath: %s", goBin)
```

Comment:
swallowed root error

Reply:
since its custom built error, `Invalid go binary: exit status` thought this error would be more apt. will add the cause to this error.

Reply:
yes i understand.
but newgoget can return another error in a future so it would be good to know root cause of this.

```go
fmt.Errorf("go get fetcher can't be initialised with gobinpath '%s': %v", goBin, err)
```